authorization/policy.go

```diff
@@ -34,23 +34,17 @@ func (p ActionResourcePolicy) Authorize(ctx context.Context, userID string, auth
 		}
 	}
 
-	if ok, err := authorizer.IsAuthorizedWithContext(ctx, userID, p.Action, resource); err != nil {
+	if ok, reason, err := authorizer.IsAuthorizedWithReasonWithContext(ctx, userID, p.Action, resource); err != nil {
 		switch status.Code(err) {
 		case codes.Canceled:
 			return context.Canceled
 		}
 
 		return fmt.Errorf("unable to call IsAuthorizedWithContext: %w", err)
 	} else if !ok {
-		if _, err := authorizer.GetResourceWithContext(ctx, resource.Id, resource.Type); err != nil {
-			switch status.Code(err) {
-			case codes.Canceled:
-				return context.Canceled
```

Comment:
I think we still need to convert the gRPC canceled into context.Canceled, to avoid the error traces we had issues with a while back.

Reply:
Nice catch!